util/types/datum.go

```
@@ -1343,14 +1309,13 @@ func ConvertDatumToDecimal(sc *variable.StatementContext, d Datum) (*MyDecimal,
 		err = sc.HandleTruncate(dec.FromString(d.GetBytes()))
 	case KindMysqlDecimal:
 		*dec = *d.GetMysqlDecimal()
-	case KindMysqlHex:
-		dec.FromInt(d.GetMysqlHex().Value)
-	case KindMysqlBit:
-		dec.FromUint(d.GetMysqlBit().Value)
 	case KindMysqlEnum:
 		dec.FromUint(d.GetMysqlEnum().Value)
 	case KindMysqlSet:
 		dec.FromUint(d.GetMysqlSet().Value)
+	case KindBinaryLiteral, KindMysqlBit:
+		val, _ := d.GetBinaryLiteral().ToInt()
```

error is not handled.